mitodl/bootcamp-ecommerce
# Fixed react-dropzone-uploader bugs

#### What are the relevant tickets?
Fixes #901 

#### What's this PR do?
- Gets rid of react warning when loading the dropzone component
- Clearly shows errors within the component, and shows error text from the server response if any came back

#### How should this be manually tested?
First, create/edit and application such that it is awaiting resume upload

Then:
- Set `MAX_RESUME_FILE_SIZE` to something like `1`, then try to upload a file. You should see an error saying the file is too large.
- Apply the patch below to force an error from the server when a resume upload is attempted, then use the dropzone to upload a resume. You should see the server error message displayed.
- Use the interactive elements of the dropzone and make sure everything looks correct.

```diff
diff --git a/applications/views.py b/applications/views.py
index a3e21130..602070df 100644
--- a/applications/views.py
+++ b/applications/views.py
@@ -193,6 +193,11 @@ class UploadResumeView(GenericAPIView):
         """
         Update the application with resume and/or linkedin URL
         """
+        return Response(
+            {"error": "This is an error from the server"},
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
         application = self.get_object()
         linkedin_url = request.data.get("linkedin_url")
         resume_file = request.FILES.get("file")
```

#### Background info
I created a fork of the react-dropzone-uploader repo under the mitodl org, made some fixes and additions, and published the package to npm ([forked repo](https://github.com/mitodl/react-dropzone-uploader), [npm page](https://www.npmjs.com/package/@mitodl/react-dropzone-uploader))

#### Screenshots (if appropriate)
<img width="553" alt="ss 2020-09-03 at 14 22 19 " src="https://user-images.githubusercontent.com/14932219/92164598-975d2d00-ee03-11ea-8033-5f43806afc08.png">
<img width="545" alt="ss 2020-09-03 at 14 25 43 " src="https://user-images.githubusercontent.com/14932219/92164597-96c49680-ee03-11ea-9700-cc0b1a24294d.png">
<img width="547" alt="ss 2020-09-03 at 14 26 56 " src="https://user-images.githubusercontent.com/14932219/92164593-95936980-ee03-11ea-85d8-5f040d73f2bc.png">
